Reference to #864 fixes for bootstrap-5 theme

ccsliinc opened the pull request:
This was a change implemented via the coreui update.

### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests and did you add any new tests needed for your feature?
2. [ ] Did you update all templates (if applicable)?
3. [ ] Did you add the [relevant documentation](https://github.com/rappasoft/laravel-livewire-tables-docs) (if applicable)?
4. Did you test locally to make sure your feature works as intended?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

ccsliinc committed:
Reference to #864 fixes for bootstrap-5 theme

This was a change implemented via the coreui update.

rappasoft commented:
I don't know whats different for me, but it works with what's there and doesn't work with this PR.

ccsliinc commented:
I don't know if I missed something.  Bulk Actions dropdown was not working.  I need to pull a latest and check on my code.  It seems the code now(or has been) correct as it matches my code.  I will test when I have a moment.

ccsliinc commented:
I double checked and the other dropdowns are working but no the Bulk Actions on BootStrap-5.  I am updating the code to follow the same method of dropdown as the column select.

I double checked and it does not work with the latest version, but this code update fixes it.

ccsliinc committed:
Updated toolbar.php to allow dropdown functionality.

Changed the dropdown behavior to Alpine.js to the same behavior as the column select.

ccsliinc commented:
Code screenshot - Not Working
<img width="1295" alt="Current Verison - Not Working" src="https://user-images.githubusercontent.com/5481458/214143875-e8e5259d-543f-4c41-9a47-3c21270174c9.png">

Code screenshot - Working
<img width="1293" alt="Updated Code - Working" src="https://user-images.githubusercontent.com/5481458/214143873-22174214-28ce-4ac8-8c84-d445938a77f7.png">

ccsliinc committed:
Update toolbar.blade.php

wire:key name was incorrect.

lrljoe commented:
Is this still an issue?

ccsliinc commented:
I need to check, give me a day.  I need to setup a test site.

ccsliinc commented:
This is still not working.

lrljoe commented:
It does seem to work without it in BS4 and BS5 on the demo environment, I do see the advantage in standardising though!  Can you confirm which version of BS5 you're using, and which plugins/modules you've got included?

ccsliinc commented:
The code for each dropdown is different.  The column select works but the actions does not.  I am using latest bootstrap along with coreui latest version.  My PR is just taking the working code in the column select and duplicating it in the actions.

lrljoe commented:
No quite right, Bulk Actions doesn't work in BS5, and this fixes it.

Apologies, when I was testing I kept thinking it was column select that was supposed to be broken.

This does need a pull @rappasoft

ccsliinc commented:
@lrljoe Thank You.

lrljoe commented:
My plan is to review this PR comprehensively tomorrow evening  to make sure I'm not missing anything.

I've got a tiny bit of tidying up to do elsewhere, and a couple of new tests to write. But I would expect to be issuing a pull to master by the end of the weekend.

Then it'll be down to when Anthony has time to cast an eye over the rather large chunk of changes that are prepped.

Going forward it should be little and more often as I've just cleared out a backlog of my own PRs as well so it'll be smoother sailing.

I really appreciate you putting the time in, and I know Anthony does too!

ccsliinc commented:
Thank you.  Always around to lend a helping hand.

lrljoe commented on resources/views/components/tools/toolbar.blade.php:
```diff
@@ -734,12 +734,28 @@ class="dropdown-item text-center"
+                         x-data="{ open: false }"
+                         x-on:keydown.escape.stop="open = false"
+                         x-on:mousedown.away="open = false"
+                         wire:key="column-select-button-{{ $component->getTableName() }}"
```
Should be bulk-actions not column-select

lrljoe commented on resources/views/components/tools/toolbar.blade.php:
```diff
@@ -768,7 +784,7 @@ class="dropdown d-block d-md-inline"
                             class="btn dropdown-toggle d-block w-100 d-md-inline"
                             type="button"
-                            id="columnSelect-{{ $component->getTableName() }}"
+                            wire:key="bulk-actions-dropdown-button-{{ $component->getTableName() }}"
```
this is column-select, not bulk-actions, so wire:key needs amending

lrljoe commented on resources/views/components/tools/toolbar.blade.php:
```diff
@@ -768,7 +784,7 @@ class="dropdown d-block d-md-inline"
                             x-on:click="open = !open"
                             class="btn dropdown-toggle d-block w-100 d-md-inline"
                             type="button"
-                            id="columnSelect-{{ $component->getTableName() }}"
```
id is required for some browsers for the aria-labelledby, so should be restored

lrljoe reviewed (changes requested):
There are some fixes required before this functions as expected, at present it breaks Column Select due to wire:key/id issues

ccsliinc committed:
Update toolbar.blade.php

Changes as requested

ccsliinc commented:
@lrljoe updated as requested.

lrljoe commented:
Thanks, looks okay to me, but I'll give it a once-over tomorrow with fresh eyes and merge it into develop as the last item for this round of pulls, as this actually fixes a bug that's fairly obstructive, and I don't feel right leaving it out.

ccsliinc commented:
Excellent, I agree.  Thank you.  Let me know if you need anything else.

lrljoe commented:
No problem, thank you for taking the time to address this!  I know Anthony really appreciates anyone taking time out to lend a hand!

Please don't be surprised by some random pulls for this tomorrow into oddly named branches, as I'll stick it into a couple of different edge cases that I have setup for testing just to make sure it doesn't impact anything, it looks okay to me, but I've been staring at a screen for about 18 hours now, so I'm not at my best.

I am also updating the Demo with a chunk of the core features so testing in future should be much easier.

lrljoe commented:
Yeah this seems to do the trick now, a little more testing needed tomorrow just to make sure nothing is conflicting anywhere.  I think that at 26 hours in front of a screen I'm not in any state to consider merging anything!  Looks good though, great job!

Your fix helpfully also doesn't impact my next proposed (relatively major) change (AlpineJS approach to Bulk Actions) which is a relief.  

Effectively I've replaced the constant binding/refreshing of the table (and re-running of the query) when using Bulk Actions, with some slightly custom AlpineJS lazy binding, that persists the checkbox state without having to constantly re-render the page, also allows for a "Select All On Page" in addition to "Select All" approach, which is a bit smoother!

Expect this to merge into develop by 1pm BST tomorrow, and then I'll issue the PR into master, and hopefully I won't break Anthony with the quantity of changes!

ccsliinc commented:
Thank you!  Get some rest and give those eyes a break.  Look forward to pulling when its complete, it will allow me to get rid of the imported views.

If you need anything let me know.  I am always around for a hand.
